server/container_create.go

```diff
@@ -393,6 +393,11 @@ func (s *Server) getAppArmorProfileName(annotations map[string]string, ctrName s
 	}
 
 	if profile == apparmor.ProfileRuntimeDefault {
```

Comment:
Essentially we should only call `EnsureDefaultApparmorProfile` if `config.ApparmorProfile == ""` (which you'd need to keep track of in the server instance).